constraint/pkg/client/e2e_test.go

```diff
@@ -0,0 +1,705 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/client/drivers/local"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/core/templates"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/types"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8schema "k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	denied    = "DENIED"
+	rejection = "REJECTION"
+	dryRun    = "DRYRUN"
+)
+
+func newConstraintTemplate(name, rego string, libs ...string) *templates.ConstraintTemplate {
+	return &templates.ConstraintTemplate{
+		ObjectMeta: metav1.ObjectMeta{Name: strings.ToLower(name)},
+		Spec: templates.ConstraintTemplateSpec{
+			CRD: templates.CRD{
+				Spec: templates.CRDSpec{
+					Names: templates.Names{
+						Kind: name,
+					},
+					Validation: &templates.Validation{
+						OpenAPIV3Schema: &apiextensions.JSONSchemaProps{
+							Type: "object",
+							Properties: map[string]apiextensions.JSONSchemaProps{
+								"expected": {Type: "string"},
+							},
+						},
+					},
+				},
+			},
+			Targets: []templates.Target{
+				{Target: "test.target", Rego: rego, Libs: libs},
+			},
+		},
+	}
+}
+
+func newConstraint(kind, name string, params map[string]string, enforcementAction *string) *unstructured.Unstructured {
+	c := &unstructured.Unstructured{}
+	c.SetGroupVersionKind(k8schema.GroupVersionKind{
+		Group:   "constraints.gatekeeper.sh",
+		Version: "v1alpha1",
```

Comment:
do we want to update apiVersions?

Reply:
Yes. There's actually a few changes I want to make related to string constants, but I'd like to have a PR just for that:

https://github.com/open-policy-agent/frameworks/issues/163